types/mongoose/index.d.ts

```diff
@@ -877,6 +882,14 @@ declare module "mongoose" {
       error: mongodb.MongoError, doc: T, next: (err?: NativeError) => void
     ) => void): this;
 
+    post<T extends Document>(method: string | RegExp, { document, query }: HookOptions, fn: (
```

Comment:
I'm not sure what extracting this to a separate interface would achieve. Seems to be identical, functionally, to the previous version.

Also, the `fn` signature was previously defined correctly, as per the mongoose documentation here:
https://mongoosejs.com/docs/middleware.html#error-handling-middleware
```js
// Handler **must** take 3 parameters: the error that occurred, the document
// in question, and the `next()` function
schema.post('save', function(error, doc, next) {
```

Could you clarify?

Reply:
Oops, I thought the commented out section below was the previous implementation, and this is what you changed it to. Disregard above comment.

It seems that the hooks have a different signature when used with `deleteOne` and `remove` as per https://mongoosejs.com/docs/middleware.html#naming

You should limit this signature to only those methods. See the lines just above your changes, which define different signatures for `insertMany`, on how to do it.

Reply:
yep makes sense, pushed new changes that include the "remove" method also.